### DISCUSSION TITLE
Unwrapping an optional presentation store leads to type mismatch.

### DISCUSSION THREAD

**Discussion by jessedoescode** [pointfreeco/swift-composable-architecture#3529] [Q&A]
Hi,

When modeling navigation with an enum:
```
@Reducer(state: .equatable)
    enum PresentEvent {
        case create(CreateEventFeature)
        case edit(EditEventFeature)
        case saved(SavedEventPresentFeature)
        case unsaved(UnsavedEventFeature)
    }
```

That is held in the feature's state:
```
@Presents var presentEvent: PresentEvent.State?
```

As well as action:
```
 case presentEvent(PresentationAction<PresentEvent.Action>)
```

When trying to unwrap it with `if let`:
```
if let store = store.scope(state: \.presentEvent, action: \.presentEvent) {
      PresentEventView(store: store)
}

struct PresentEventView: View {
    
    var store: StoreOf<EventsFeature.PresentEvent>
   .....
```

I get a type mismatch due to the `PresentationAction`:

> Cannot convert value of type 'Store<EventsFeature.PresentEvent.State, PresentationAction<EventsFeature.PresentEvent.Action>>' to expected argument type 'StoreOf<EventsFeature.PresentEvent>' (aka 'Store<EventsFeature.PresentEvent.State, EventsFeature.PresentEvent.Action>')

**Comment by mbrandonw** [comment]
Hi @jessedoescode, the error is telling you the problem. The store you have constructed has the action `PresentationAction<EventsFeature.PresentEvent.Action>`, but you are passing it to a view that wants a store with the action `EventsFeature.PresentEvent.Action`. Notice the extra layer of `PresentationAction<…>`.

To peel away that layer you can further chain on `.presented`, which is the [case inside `PresentationAction`](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/presentationaction/presented(_:)) that holds onto your type:

```swift
if let store = store.scope(state: \.presentEvent, action: \.presentEvent.presented) {
  …
}
```

**Comment by jessedoescode** [comment]
@mbrandonw Ahh. Yep, error message is super clear! Just hadn't come across the `.presented` case like this. Thanks for the quick reply as always and appreciate the hard work!
